samples/features/sql-big-data-cluster/connectivity/webhdfs-java-client/src/main/java/com/microsoft/mssql/App.java

```diff
@@ -0,0 +1,118 @@
+	// Properties to be configured before running the application.
+	private static String GW_ENDPOINT = "https://knox.test.azdata.local:30443/gateway/default/webhdfs/v1";
+	
+	private static String KDC = "winad.azdata.local";
+	
+	private static String DOMAIN_REALM = "AZDATA.LOCAL";
+	
+	// The existing login.conf along with this project can be modified instead of providing a new path.
+	private static String LOGIN_CONF_PATH = "login.conf";
+	// End of properties to be configured.
+
+	static Credentials emptyCredentials = new Credentials() {
+		public String getPassword() {
+			return null;
+		}
+		public Principal getUserPrincipal() {
+			return null;
+		}
+	};
+	
+	static CredentialsProvider provider = new CredentialsProvider() {
+		@Override
+		public void setCredentials(AuthScope authscope, Credentials credentials) {
+			// No op
+		}
+
+		@Override
+		public Credentials getCredentials(AuthScope authscope) {
+			//TODO: this can be enhanced to pass controller use name and password in case the AuthScope authscheme is basic.
+			return emptyCredentials;
+		}
+
+		@Override
+		public void clear() {
+			// No op. Nothing to clean up.
+		}
+	};
+        private static void uploadResource(String inputFilePath, String outputFilePath){
+        	System.out.println("Entering create resource");
+                // Construct Knox endpoint
+                String createOperationEndpoint = GW_ENDPOINT + outputFilePath + "?op=CREATE&overwrite=true"; 
+	        try (CloseableHttpClient client2 = HttpClients.custom().setDefaultCredentialsProvider(provider).build()) {
+                        HttpUriRequest request = new HttpPut(createOperationEndpoint);
+                        // First request to get the location in data nodes
+                        try (CloseableHttpResponse response = client2.execute(request)) { 
+                                System.out.println("===============");
+                                String newlocation = response.getFirstHeader("Location").getValue();
+                                HttpEntity entity = response.getEntity();
+                                System.out.println("----------------------------------------");
+                                System.out.println(response.getStatusLine());
+                                System.out.println("----------------------------------------");
+                                if (entity != null) {
+                                        System.out.println(EntityUtils.toString(entity));
+                                }
+                                System.out.println("----------------------------------------");
+                                EntityUtils.consume(entity);
+                                // Second request to put the content to that location
+                                File testUploadFile = new File(inputFilePath);
+                                HttpEntity putData = MultipartEntityBuilder.create().addBinaryBody("upfile", testUploadFile).build();
```

Comment:
Is `upFile` the name of the file that will be created in HDFS ? If yes, then this should probably be a command line arg.

Reply:
No. It is the body name. Looks like it is not matter.